library/core/src/pin.rs

```diff
@@ -802,6 +802,44 @@ impl<T: ?Sized> Pin<&'static T> {
     }
 }
 
+impl<'a, P: DerefMut> Pin<&'a mut Pin<P>> {
+    /// Gets a pinned mutable reference from this nested pinned pointer.
+    ///
+    /// This is a generic method to go from `Pin<&mut Pin<Pointer<T>>>` to `Pin<&mut T>`. It is
+    /// safe because the existence of a `Pin<Pointer<T>>` ensures that the pointee, `T`, cannot
+    /// move in the future, and this method does not enable the pointee to move. "Malicious"
+    /// implementations of `Pointer::DerefMut` are likewise ruled out by the contract of
+    /// `Pin::new_unchecked`.
+    #[unstable(feature = "pin_deref_mut", issue = "none")]
```

Comment:
So this is marked unstable but also insta-stably exposed already through the `Future` impl? That seems odd to me. It defeats the point of an unstable function, namely that we can remove it again if something turns out to be wrong with it.

Reply:
This function here cannot be removed again without incurring a breaking change. The more honest thing thus would be to make it insta-stable. Therefore it should go through T-libs FCP.

Reply:
The ability to do this operation would be insta-stable, but not e.g. the method name or other details that the libs team might want to modify. Would love to get thoughts from @rust-lang/libs if they have any, though.

Reply:
I'm not particularly worried about needing to rename this from `as_deref_mut`, though we can nominate it for discussion in the next libs meeting to make sure everyone's on the same page. Do you forsee any details that we might want to change?

Reply:
Can you [open a tracking issue](https://github.com/rust-lang/rust/issues/new?assignees=&labels=C-tracking-issue%2C+T-libs&template=library_tracking_issue.md&title=Tracking+Issue+for+XXX) for this?

Reply:
Ah, will open later today or tomorrow!

Reply:
Done: https://github.com/rust-lang/rust/issues/86918.